draft-ietf-quic-transport.md

```diff
@@ -3592,6 +3594,21 @@ and will contain a CRYPTO frame with an offset matching the size of the CRYPTO
 frame sent in the first Initial packet.  Cryptographic handshake messages
 subsequent to the first do not need to fit within a single UDP datagram.
 
+
+### Abandoning Initial Packets {#discard-initial}
+
+A client stops both sending and accepting Initial packets when it sends its
+first Handshake packet.  A server stops sending and accepting Initial packets
+when it receives its first Handshake packet.  Though packets might still be in
+flight or awaiting acknowledgment, no further Initial packets need to be
+exchanged beyond this point.  Initial packet protection keys are discarded (see
+Section 4.10 of {{QUIC-TLS}}) along with any loss recovery and congestion
+control state (see Sections 5.3.1.2 and 6.9 of {{QUIC-RECOVERY}}).
```

Comment:
Not my area, but I'm not sure implies the right thing to say that any congestion control state gets discarded here.  The congestion controller doesn't care what encryption level things are at, it just cares about bytes.

Perhaps it would be worth being more concrete:  "discarded(...), and all Initial packets are considered acknowledged for loss recovery and congestion control purposes (see...)."  This would also make the following paragraph unnecessary.

Reply:
+1. @MikeBishop 's proposed text also explicitly articulates that this is an implicit acknowledgement, something that we've been shy to do before.

Reply:
I don't think that it is the case.  We don't really say anything from a loss recovery perspective - the packets are not declared lost.  The congestion controller is just told to remove the bytes from in flight - they don't cause the CWND to increase.  So it's not really acknowledgment.

As the changes to the recovery draft state: loss recovery state is just thrown out: you won't be retransmitting anything at the encryption level.  So saying anything more isn't really helpful there.  The effect on the congestion controller is intentionally neutral: don't increase CWND as though the packets got through, but don't keep the bytes on the books as though the packets were lost.

Or have I screwed this up?

Reply:
Okay, I see that reading of the -recovery text.  But that's really weird -- we *do* know that these packets have arrived, so why define a special operation for "forgetting" packets rather than just marking them arrived as normal and be done?

If one component were trying to game the other -- for example, by sending many Initial packets with redundant data -- then you wouldn't want to let the packetizer force many packets to be "acknowledged" when only one actually needed to get through to produce the effect.  But the ways to game that I see offhand are intra-implementation, and I think we can say that's not an attack of concern.

Reply:
Good points. I'd say the conservative approach is the one @martinthomson suggests. I wonder if it might be better to be more explicit about it though, but I'll save that for another PR I think.